src/clients/identity/types.ts

```diff
@@ -0,0 +1,92 @@
+export enum AuthHeaderType {
+    JWT = 'JWT',
+    AccessToken = 'AccessToken'
+}
+
+export interface AuthHeader {
+    token: string;
+    type: AuthHeaderType;
+}
+
+export interface IWithAuthenticationOptions {
```

Comment:
why is the withAuthenticationOptions exported from here?
I'm guessing to avoid circular dependency, so maybe rename the interface, and have WithAuthenticationOptions extend from it in with-authentication.ts